SmartEVSE-3/src/main.cpp

```diff
@@ -5616,10 +5632,42 @@ bool fwNeedsUpdate(char * version) {
     return false;
 }
 
+/**
+ * Periodically retrieves current measurements from the HomeWizard P1 energy meter
+ * and updates the main meter's currents.
+ *
+ * This function checks if the main meter type is set to HomeWizard P1 (EM_HOMEWIZARD_P1)
+ * and ensures a delay of at least 5 seconds between consecutive data retrieval attempts.
+ */
+void homewizard_loop() {
+    static unsigned long lastCheck_homewizard = 0;
+
+    constexpr unsigned long interval = 5000; // 5 seconds
+    const unsigned long currentTime = millis();
+
+    if (MainsMeter.Type != EM_HOMEWIZARD_P1 || (currentTime - lastCheck_homewizard < interval)) {
+        return;
+    }
+
+    _LOG_A("homewizard_loop(): start HomeWizrd P1 reading.");
+
+    lastCheck_homewizard = currentTime;
+
+    const auto currentsP1 = getMainsFromHWP1();
+
+    if (currentsP1.first) {
+        const auto l1a = static_cast<int16_t>(currentsP1.second[0] * 10);
+        const auto l2a = static_cast<int16_t>(currentsP1.second[1] * 10);
+        const auto l3a = static_cast<int16_t>(currentsP1.second[2] * 10);
+        setMainsMeterCurrents(l1a, l2a, l3a);
+    }
+}
 
 void loop() {
 
     network_loop();
+    homewizard_loop();
```

Comment:
I would prefer:
if (MainsMeter.Type == EM_HOMEWIZARD_P!)
     homewizard_loop();

..so developers don't need to look into homewizard_loop code if they are not interested in the homewizard code .

It would change line 5648 too of course...